api/python/provisioner/commands/deploy_dual.py

```diff
@@ -189,10 +188,18 @@ def _run_states(self, states_group: str, run_args: run_args_type):
                     self._apply_state(
                         f"components.{state}", secondaries, stages
                     )
+                elif state in (
+                    "ha.cortx-ha",
+                ):
+                    # Execute first on primary then on secondary sequentially.
+                    nodes = Deploy._get_node_list()
+                    self._apply_state(f"components.{state}", primary, stages)
+                    if primary in nodes:
+                        nodes.remove(primary)
```

Comment:
if condition might not be necessary here.
Primary would always be present, even in case of a single node. So this condition shall always be true.